gptme/server/api_v2.py

```diff
@@ -346,6 +358,19 @@ def step(
         lock=False,
     )
 
+    # TODO: This is not the best way to manage the chdir state, since it's
+    # essentially a shared global across chats (bad), but the fix at least
+    # addresses the issue where chats don't start in the directory they should.
+    # If we are attempting to make a step in a conversation without any user
+    # messages, make sure we first chdir to the workspace directory (so that
+    # the conversation starts in the right folder).
+    user_messages = [msg for msg in manager.log.messages if msg.role == "user"]
+    if not user_messages:
+        logger.debug(
+            f"No user messages found, changing directory to workspace: {workspace}"
+        )
+        os.chdir(workspace)
```

Comment:
Fixes an annoying bug, but doesn't fix the underlying issue with multiple chats managing the same pwd state.

Reply:
Could be the same issue as https://github.com/gptme/gptme/issues/594, in that case https://github.com/gptme/gptme/pull/595 should have fixed it.